omod/src/main/webapp/resources/scripts/components/LabTrackingOrderFactory.js

```diff
@@ -390,12 +405,49 @@ angular.module("labTrackingOrderFactory", [])
             _updateObsIfExists(labTrackingOrder, "accessionNumber", obs, obsIdsToDelete);
             _updateObsIfExists(labTrackingOrder, "urgentReview", obs, obsIdsToDelete);
 
-            if (labTrackingOrder.postopDiagnosis.diagnosis.value != null && labTrackingOrder.postopDiagnosis.diagnosis.value.length > 0) {
-
+            // special case for post-op diagnosis obs group
+            if (((typeof labTrackingOrder.postopDiagnosis.diagnosis === 'object' && labTrackingOrder.postopDiagnosis.diagnosis.value === null)
+                || labTrackingOrder.postopDiagnosis.diagnosis === null || labTrackingOrder.postopDiagnosis.diagnosis.length === 0)) {
+                // if no post-op diagnosis specified, but there's an existing value, we need to delete the whole obs group
+                if (labTrackingOrder.postopDiagnosis.groupMemmberParentUuid !== null) {
+                  obsIdsToDelete.push(labTrackingOrder.postopDiagnosis.groupMemmberParentUuid);
+                }
+            }
+            else {
+                // a post-op diagnosis value has been entered
                 var v = [];
-                v.push(Encounter.toObsWebServiceObject(LabTrackingOrder.concepts.postopDiagnosis.value,
-                    labTrackingOrder.postopDiagnosis.diagnosis.value,
-                    labTrackingOrder.postopDiagnosis.obsUuid));
+
+                if (typeof labTrackingOrder.postopDiagnosis.diagnosis === 'object') {
+                  // coded obs
+                  if (labTrackingOrder.postopDiagnosis.obsUuid === null || labTrackingOrder.postopDiagnosis.type === 'coded') {
+                    // if there's no existing obs (obsUuid === null) or the existing type was coded, just create/update
+                    v.push(Encounter.toObsWebServiceObject(LabTrackingOrder.concepts.postopDiagnosis.value,
+                      labTrackingOrder.postopDiagnosis.diagnosis.value,
+                      labTrackingOrder.postopDiagnosis.obsUuid));
+                  }
+                  else {
```

Comment:
Wait wait wait is there a scenario where `postopDiagnosis.diagnosis` is a non-null object but `postopDiagnosis.type` is not `coded??

Reply:
Going to wait to review the rest until I understand what's going on here, @mogoodrich

Reply:
What's happening here is that we've changed the form field for postopDiagnosis from a typeahead that doesn't allow free-text entry to one that does (see changes in labtrackingOrderDetails-specimen.gsp) ... "postopDiagnosis.diagnosis" is the element that backs this field, so it could end up being a display/value pair or a string depending on what is entered. 

Depending on which, we want to use a different "question" concept (Diagnosis Coded concept or Diagnosis Non-Coded concept).  In the use case in which the user updates the field and changes from coded to free-text (or vice versa) it was running into an error due to trying to change the concept associated with an existing obs.  So I added "type", so that in the case of an edit, we store the type of the originally loaded obs returned from the server... and we use that to determine if upon submit we need just update the obs, or void and create a new obs.

Reply:
Got it, that makes sense. I suggest calling it something like `originalType` then.

Reply:
Agree, makes sense @brandones  I'll update...